sklearn/utils/_param_validation.py

```diff
@@ -114,8 +114,8 @@ def make_constraint(constraint):
     """
     if isinstance(constraint, str) and constraint == "array-like":
         return _ArrayLikes()
-    if isinstance(constraint, str) and constraint == "sparse matrix":
-        return _SparseMatrices()
+    if isinstance(constraint, str) and constraint == "sparse container":
+        return _SparseContainers()
```

Comment:
We need to change all the occurrences of `"sparse matrix"` to `"sparse container"` for input parameters' validation. Doc-strings also need to be changed accordingly.